pkg/log/log.go

```diff
@@ -2,177 +2,162 @@ package log
-// the given key/value pairs as context.
-func (p *Logger) Info(ctx *gin.Context, msg string) *Logger {
-	return p.Infof(ctx, msg)
+// WithValues creates a new logger with additional key-value pairs in the context
+func WithValues(keysAndValues map[string]interface{}) *Logger {
+	return logger.WithValues(keysAndValues)
 }
 
-// Infof logs are used for non-error messages. It will log a message with
-// the given key/value pairs as context.
-func (p *Logger) Infof(ctx *gin.Context, msg string, args ...string) *Logger {
+// Info logs a non-error message.
+func (l *Logger) Info(ctx *gin.Context, msg string) {
+	Infof(ctx, msg)
+}
+
+// Infof logs a non-error formatted message.
+func (l *Logger) Infof(ctx *gin.Context, msg string, args ...string) {
 	ctxInfo := addContextInfo(ctx)
 	arguments := make([]interface{}, len(args))
 	for i, arg := range args {
 		arguments[i] = arg
 	}
 	if len(arguments) > 0 {
-		p.lgr.Info(fmt.Sprintf(msg, arguments...), ctxInfo...)
+		l.logr.Info(fmt.Sprintf(msg, arguments...), ctxInfo...)
 	} else {
-		p.lgr.Info(msg, ctxInfo...)
+		l.logr.Info(msg, ctxInfo...)
 	}
-
-	return p
 }
 
-// Error logs are used for logging errors. It will log the error with the given
-// message and key/value pairs as context.
-func (p *Logger) Error(ctx *gin.Context, err error, msg string) *Logger {
-	return p.Errorf(ctx, err, msg)
+// Error logs the error with the given message.
+func (l *Logger) Error(ctx *gin.Context, err error, msg string) {
+	l.Errorf(ctx, err, msg)
 }
 
-// Errorf logs are used for logging errors. It will log the error with the given
-// message and key/value pairs as context.
-func (p *Logger) Errorf(ctx *gin.Context, err error, msg string, args ...string) *Logger {
+// Errorf logs the error with the given formatted message.
+func (l *Logger) Errorf(ctx *gin.Context, err error, msg string, args ...string) {
 	ctxInfo := addContextInfo(ctx)
 	arguments := make([]interface{}, len(args))
 	for i, arg := range args {
 		arguments[i] = arg
 	}
 
 	if len(arguments) > 0 {
-		p.lgr.Error(err, fmt.Sprintf(msg, arguments...), ctxInfo...)
+		l.logr.Error(err, fmt.Sprintf(msg, arguments...), ctxInfo...)
 	} else {
-		p.lgr.Error(err, msg, ctxInfo...)
+		l.logr.Error(err, msg, ctxInfo...)
 	}
-
-	return p
 }
 
-// WithValues appends tags to the logger.
-func (p *Logger) WithValues(keysAndValues ...interface{}) *Logger {
+// WithValues creates a new logger with additional key-value pairs in the context
+func (l *Logger) WithValues(keysAndValues map[string]interface{}) *Logger {
 	if len(keysAndValues) > 0 {
-		tags := append([]interface{}(nil), p.tags...)
-		tags = append(tags, keysAndValues...)
-		p.tags = tags
-		// ZapLoggerTo, WithName and WithValues all return new logger instances.
-		// The logger must be set again with the values stored in the Logger struct.
-		p.lgr = logf.ZapLoggerTo(p.out, p.isTestingMode).WithName(p.name).WithValues(tags...)
+		nl := newLogger(logger.name)
```

Comment:
calling this would be redundant because the WithValues would return a new logger which overwrites this one here.